### PR TITLE
Initial Dragonflight Update

### PR DESCRIPTION
**Pull request by rookdorf**
This contains some basic fixes for 10.0 compatibility as well as an initial simple implementation of an Essence ring. It works fine, but doesn't show in progress Essence generation yet, and reuses Chi's color. I may return to it and improve the ring by showing regenerating Essence progress soon, though honestly I'm hoping for API updates that make it cleaner and easier to do so. I will update this PR in place if it hasn't been merged yet.

### CODE REVIEW

**Comment by rherzfeld** [comment]
I am still getting an error, when the fatigue or breath timer should show up.
```
Interface/AddOns/ArcHUD3/Rings/MirrorTimer.lua:147: attempt to index global 'MirrorTimerColors' (a nil value)
[string "@Interface/AddOns/ArcHUD3/Rings/MirrorTimer.lua"]:147: in function `?'
[string "@Interface/AddOns/ArcHUD3/Libs/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:119: in function <...UD3/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:119>
[string "=[C]"]: ?
[string "@Interface/AddOns/ArcHUD3/Libs/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:29: in function <...UD3/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:25>
[string "@Interface/AddOns/ArcHUD3/Libs/CallbackHandler-1.0-7/CallbackHandler-1.0.lua"]:64: in function `Fire'
[string "@Interface/AddOns/ArcHUD3/Libs/AceEvent-3.0-4/AceEvent-3.0.lua"]:120: in function <...ce/AddOns/ArcHUD3/Libs/AceEvent-3.0/AceEvent-3.0.lua:119>
```

**Comment by rookdorf** [comment]
Good catch, I'll check on that and update this soon!

**Comment by rookdorf** [comment]
@rherzfeld Just pushed a fix for this. If you're just copying changes to your local files, it should just be a few things in MirrorTimer.lua to grab. Thanks for the heads up!

**Review by rherzfeld — COMMENTED** [review]
While checking the German translation, I noticed two issues with the localizations.

Also, I activated some arcs for testing purposes and when turning on the focus power arc, I got the following error:
 ```
 Interface/AddOns/ArcHUD3/Rings/FocusPower.lua:56: attempt to call global 'UnitMana' (a nil value)
[string "@Interface/AddOns/ArcHUD3/Rings/FocusPower.lua"]:56: in function `OnModuleUpdate'
[string "@Interface/AddOns/ArcHUD3/ModuleCore.lua"]:233: in function `ARCHUD_MODULE_UPDATE'
[string "@Interface/AddOns/ArcHUD3/ModuleCore.lua"]:137: in function <Interface/AddOns/ArcHUD3/ModuleCore.lua:112>
[string "=[C]"]: ?
[string "@Interface/AddOns/ArcHUD3/Libs/AceAddon-3.0-13/AceAddon-3.0.lua"]:70: in function <...ce/AddOns/ArcHUD3/Libs/AceAddon-3.0/AceAddon-3.0.lua:65>
[string "@Interface/AddOns/ArcHUD3/Libs/AceAddon-3.0-13/AceAddon-3.0.lua"]:527: in function <...ce/AddOns/ArcHUD3/Libs/AceAddon-3.0/AceAddon-3.0.lua:520>
[string "=(tail call)"]: ?
[string "@Interface/AddOns/ArcHUD3/ModuleCore.lua"]:689: in function <Interface/AddOns/ArcHUD3/ModuleCore.lua:686>
[string "=[C]"]: ?
[string "@Interface/AddOns/ArkInventory/Externals/Ace3/AceConfig/AceConfigDialog-3.0-85/AceConfigDialog-3.0.lua"]:45: in function <...ceConfig/AceConfigDialog-3.0/AceConfigDialog-3.0.lua:43>
[string "@Interface/AddOns/ArkInventory/Externals/Ace3/AceConfig/AceConfigDialog-3.0-85/AceConfigDialog-3.0.lua"]:827: in function <...ceConfig/AceConfigDialog-3.0/AceConfigDialog-3.0.lua:649>
[string "=[C]"]: ?
[string "@Interface/AddOns/ArcHUD3/Libs/AceGUI-3.0-41/AceGUI-3.0.lua"]:72: in function <...erface/AddOns/ArcHUD3/Libs/AceGUI-3.0/AceGUI-3.0.lua:70>
[string "@Interface/AddOns/ArcHUD3/Libs/AceGUI-3.0-41/AceGUI-3.0.lua"]:306: in function `Fire'
[string "@Interface/AddOns/ArcHUD3/Libs/AceGUI-3.0-41/widgets/AceGUIWidget-CheckBox.lua"]:68: in function <...D3/Libs/AceGUI-3.0/widgets/AceGUIWidget-CheckBox.lua:57>
 ```

File: Locales/deDE.lua [hunk 1]
```diff
@@ -157,6 +157,7 @@ L["TOOLTIP"] = {
 	CHICOMBO	= "Schaltet Anzeige von Chi (Mönch) als Kombopunkte ein/aus (die Anzeige von Kombopunkten muss aktiviert sein)",
 	RUNECOMBO 	= "Schaltet Anzeige von Runen (Todesritter) als Kombopunkte ein/aus (die Anzeige von Kombopunkten muss aktiviert sein)",
 	SOULFRAGMENTCOMBO = "Schaltet Anzeige von Seelenfragmenten (Dämonenjäger) als Kombopunkte ein/aus (die Anzeige von Kombopunkten muss aktiviert sein)",
+	RUNECOMBO 	= "Schaltet Anzeige von Essenz (Rufer) als Kombopunkte ein/aus (die Anzeige von Kombopunkten muss aktiviert sein)",
```
[the remark this answers is not in the document]
Comment (rookdorf) [reply Locales/deDE.lua]: Thanks for the sloppy copy paste fixes here

**Comment by rookdorf** [comment]
Fixed. Apparently that was changed in 3.0, so not sure how it was still around in here.

**Comment by rherzfeld** [comment]
Thank you, it looks all good to me, now

**Comment by eiymba** [comment]
wow @rookdorf, fantastic work! On a whim, I decided to resurrect this project for a friend, and I want to contact @nyyr to see if we could collaborate on the repo and get your PRs merged. I've taken the liberty of checking out your changes into my own fork [here](https://github.com/eiymba/ArcHUD3/tree/dragonflight).

Note that, my fork will have some additional bells and whistles, like prettier formatting, and AceAddons is now a submodule of the repository.

My idea is to create Github actions which will do automatic releases and version bumps in line with WoW versions. If you're interested in working together, let me know! 😄

**Comment by rookdorf** [comment]
@eiymba Hey definitely! I did try to contact nyyr actually, since my Wrath PR has sat here for 2 months I tracked down an email and tried that but haven't heard anything back yet (it's been ~2 weeks now, so not holding my breath). I was considering spinning my updates off into some forked addon and implementing GitHub actions as releases too, since I see Travis-CI is discontinued. Definitely down to merge ours

**Comment by eiymba** [comment]
@rookdorf That's such a shame. I made a cheeky release merging your changes with mine on GitHub (yours were a lot more comprehensive!). Maybe it's best we do create a spin off project. I'll add you as collaborator to my fork now and tomorrow we can figure out the details on how to proceed with a new project. Sounds good with you? 

My socials are the same username.

**Comment by rookdorf** [comment]
Sure, sounds good to me

**Comment by eiymba** [comment]
added

**Comment by eiymba** [comment]
@rookdorf pinged you on your social media to discuss a spin off project/repo if you're still down. 😄

**Comment by phoenxis4r** [comment]
> @rherzfeld Just pushed a fix for this. If you're just copying changes to your local files, it should just be a few things in MirrorTimer.lua to grab. Thanks for the heads up!

Thanks for working on this. I've used this addon since BC I think. I missed it a lot till I found your work on it.

**Comment by rookdorf** [comment]
Edit 7/23/23: Fixed the new MirrorTimer stuff in 10.1.5 that broke the breath/fatigue/feign death bars. If looking to use this completely, make sure to also rebuild it with updated libraries to avoid essence errors.

**Pull request merged**